src/plugins/kubectlGeneration/aksDocsRag/client.ts

```diff
@@ -0,0 +1,114 @@
+import { ReadyAzureSessionProvider } from "../../../auth/types";
+import { Errorable, failed, getErrorMessage } from "../../../commands/utils/errorable";
+import { aksDocsRAGEndpoint, aksDocsRAGIntent, aksDocsRAGMessagePropmt, aksDocsRAGScenario, aksDocsRAGScopes, aksDocsRAGTenantId } from "./constants";
+
+export function getAKSDocsRAGClient(sessionProvider: ReadyAzureSessionProvider) {
+    return new AKSDocsRAGClient(sessionProvider);
+}
+
+type RequestConfig = {
+    message: string;
+};
+
+export interface CommandResponse {
+    status: string;
+    message: string;
+    code: string;
+}
+
+export class AKSDocsRAGClient {
+    private authToken: string;
+    private tokenExpiresAt: number;
+    private sessionProvider: ReadyAzureSessionProvider;
+    private authScopes: string[];
+
+    constructor(sessionProvider: ReadyAzureSessionProvider) {
+        this.authToken = "";
+        this.tokenExpiresAt = 0;
+        this.sessionProvider = sessionProvider
+        this.authScopes = aksDocsRAGScopes;
+    }
+
+    private async fetchToken() {
+        const token = await this.sessionProvider.getAuthSession({ scopes: this.authScopes });
```

Comment:
can you wrap this in try/catch block to properly handle exception please?

Reply:
```
private async fetchToken(): Promise<void> {
        try {
            const token = await this.sessionProvider.getAuthSession({ scopes: this.authScopes });
            // Assuming token has properties 'token' and 'expiresAt'
            this.authToken = token.token;
            this.tokenExpiresAt = token.expiresAt;
        } catch (error) {
            console.error("Failed to fetch token:", error);
            throw new Error("Authentication failed");
        }
    }
    ```

Reply:
This has been updated, thanks